src/components/composite/Header.js

```diff
@@ -0,0 +1,359 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Link as GatsbyLinkWithoutEffects } from 'gatsby';
+import GitHubButton from 'react-github-button';
+import 'react-github-button/assets/style.css';
+import {
+  Icon,
+  Link,
+  styles,
+  Subheading,
+  TooltipLinkList,
+  WithTooltip,
+} from '@storybook/design-system';
+import GatsbyLink from '../basics/GatsbyLink';
+import Logo from '../basics/Logo';
+
+const { color, spacing, pageMargins, breakpoint, typography } = styles;
+
+const LogoWrapper = styled(Logo)`
+  && {
+    height: 22px;
+    width: auto;
+    display: block;
+    transition: transform 150ms ease-out;
+    transform: translate3d(0, 0, 0);
+    margin-top: -6px;
+
+    &:hover {
+      transform: translate3d(0, -1px, 0);
+    }
+
+    &:active {
+      transform: translate3d(0, 0, 0);
+    }
+
+    @media (min-width: ${breakpoint}px) {
+      height: 28px;
+    }
+  }
+`;
+
+const navBreakpoint = breakpoint * 1.2;
+// prettier-ignore
+const NavItem = styled.div`
+  display: inline-flex;
+  height: 3rem;
+  vertical-align: top;
+  align-items: center;
+  font-size: ${typography.size.s2}px;
+  font-weight: ${typography.weight.bold};
+  line-height: 19px;
+
+  ${props => props.showDesktop && `
+    display: none;
+    @media (min-width: ${navBreakpoint}px) {
+      display: inline-flex;
+    }
+  `}
+
+  ${props => props.showMobile && `
+    @media (min-width: ${navBreakpoint}px) {
+      display: none;
+    }
+  `}
+`;
+
+const NavTextLinkInternal = styled(GatsbyLink)`
+  && {
+    color: ${props => (props.inverse ? color.lightest : color.darkest)};
+
+    svg {
+      margin-right: 0;
+
+      path {
+        fill: ${props => (props.inverse ? color.lightest : color.darkest)};
+      }
+    }
+  }
+`;
+
+const NavTextLinkExternal = styled(Link)`
+  color: ${props => (props.inverse ? color.lightest : color.darkest)};
+  transition: transform 150ms ease-out;
+`;
+
+const NavGroup = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  z-index: 1;
+  display: flex;
+
+  ${props =>
+    props.withRightAlignment &&
+    `
+    left: auto;
+    right: 0;
+  `}
+
+  ${NavItem} + ${NavItem} {
+    margin-left: ${spacing.padding.large}px;
+  }
+`;
+
+const MobileMenuNavItem = styled(NavItem)`
+  order: 1;
+  margin-right: 10px;
+
+  > span: focus {
+    outline: none;
+  }
```

Comment:
@domyen heads up ☝️ -- I removed this for consistency w/ the old style. Had to make an assumption for now. Feel free to remove if you are ok w/ the button focus.